integration-libs/s4-service/order/components/guards/service-order.guard.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { Router, UrlTree } from '@angular/router';
+import { SemanticPathService } from '@spartacus/core';
+import { OrderDetailsService } from '@spartacus/order/components';
+import { map, Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ServiceOrderGuard {
+  constructor(
+    protected orderDetailsService: OrderDetailsService,
+    protected router: Router,
+    protected semanticPathService: SemanticPathService
+  ) {}
```

Comment:
`major:` For DI, let's use the `inject` function instead of a constructor to avoid introducing unnecessary breaking changes. See: https://wiki.one.int.sap/wiki/display/spar/Coding+Guidelines#CodingGuidelines-Useinject()functioninsteadofconstructorforDependencyInjection